recipes/gnu-config/all/test_package/conanfile.py

```diff
@@ -1,22 +1,28 @@
-from conans import ConanFile, tools
-from conans.errors import ConanException
+from conan import ConanFile
+from conan.errors import ConanException
+from conans import tools as tools_legacy
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "arch", "compiler", "build_type"
+    generators = "VirtualBuildEnv"
+    test_type = "explicit"
+    win_bash = True
 
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
     def build_requirements(self):
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+        self.tool_requires(self.tested_reference_str)
+        if self._settings_build.os == "Windows" and \
+           not self.conf.get("tools.microsoft.bash:path", default=False, check_type=bool):
```

Comment:
```suggestion
           not self.conf.get("tools.microsoft.bash:path", default=False, check_type=str):
```

bash path is a string